implementation_guides/javascript.md

```diff
@@ -2,13 +2,119 @@
 
 [Pact JS](https://github.com/pact-foundation/pact-js) is currently compliant to Pact Specification Version 2.0.
 
-Head to the [Pact JS website](https://github.com/pact-foundation/pact-js) to get started with Pact and JavaScript.
+## Quickstart guide (Consumer)
 
-## Examples
+For a Consumer using Jest and node (for example, an app created with `create-react-app`):
+
+First, you'll need to add the Pact library:
+
+    npm install --save-dev '@pact-foundation/pact'
+
+Then, add the following script to your `package.json`:
+
+      scripts: {
+        ...
+        "pactTest":
+        "export NODE_ENV=pactTest && jest --testRegex \"/*(.test.pact.js)\" --runInBand --setupFiles ./pactSetup.js --setupTestFrameworkScriptFile ./pactTestWrapper.js"
+      }
+
+You'll also need to give Pact some metadata about your setup. This is done with `pactSetup.js`:    
+
+    const path = require('path');
+    const { Pact } = require('@pact-foundation/pact');
+
+    global.port = 8989;
+    global.provider = new Pact({
+      port: global.port,
+      log: path.resolve(process.cwd(), 'logs', 'mockserver-integration.log'),
+      dir: path.resolve(process.cwd(), 'pacts'),
+      spec: 2,
+      cors: true,
+      pactfileWriteMode: 'update',
+      consumer: /* the name of your consumer */,
+      provider: /* the name of your provider */
+    });
+
+ Global setup for Jest is handled with `pactTestWrapper.js`:
+
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000; //* This is to give the pact mock server time to start
+
+    beforeAll(done => {
```

Comment:
Can this not just be `beforeAll(() => provider.setup())`?

Reply:
It can! Changed.